### DISCUSSION TITLE
Automation: Query Rows in BB Db

### DISCUSSION THREAD

**Discussion by ttelik** [Budibase/budibase#4643] [Help]
I'm having trouble implementing a seemingly simple query of an internal BB database using the 'Query Rows' step in an Automation.

I have created a trigger field to accept the search term from the web app. As you can see below, I'm filtering based on equality between this term and a column of my db.

![image](https://user-images.githubusercontent.com/74260856/155220722-1a9e1b0a-9c0b-447f-be4e-1aebd77dfc16.png)

And the result is a blank row, even though I am querying using a value present in the db. If it matters, the column I'm querying is a text field with a limit of 5 characters. My query is 4 characters.

Automation Step Input:
```
{
  "tableId": "ta_84d9618a56b74ecfbe2e852e0395d162",
  "filters": {
    "string": {},
    "fuzzy": {},
    "range": {},
    "equal": {
      "SWO": "1235",
      "tableId": "ta_84d9618a56b74ecfbe2e852e0395d162"
    },
    "notEqual": {},
    "empty": {},
    "notEmpty": {},
    "contains": {},
    "notContains": {}
  },
  "filters-def": [
    {
      "id": "1kyqIAlnw",
      "field": "SWO",
      "operator": "equal",
      "value": "1235",
      "valueType": "Binding",
      "type": "string",
      "noValue": false
    }
  ],
  "sortColumn": "SWO",
  "sortOrder": "ascending"
}
```

Automation step output:
```
{
  "rows": [],
  "success": true
}
```

And as you can see, the 1235 entry is present in my db:
![image](https://user-images.githubusercontent.com/74260856/155221520-fff348f5-ac97-4223-9326-bf59a096f62e.png)

I'm performing this test within the automation workflow editor.

**Comment by mjashanks** [comment]
Hey @ttelik - i can recreate your issue - thanks for this, looks like a bug.

A possible workaround - use `Starts With` instead of equals. This works for me, though might open up unwanted scenarios, depending on how you are using this.

**Comment by mjashanks** [comment]
I just realised what my issue was - it probably applies to you as well!

I had not set the `Limit`, `Sort Column`or `Sort order`. It shouldn't have allowed me to run the test without setting these. but once I put the values in, it worked

**Reply by ttelik** [reply]
Hi Michael,

No dice for me. Perhaps there's something I'm doing wrong. Here are the settings.

![image](https://user-images.githubusercontent.com/74260856/155327424-285be494-c9d9-4f84-b919-fdc2c988ae73.png)

**Reply by ttelik** [reply]
I can also confirm that 'starts with' works, though I'm looking for an exact match.

**Reply by ttelik** [reply]
I am also experiencing instances of the app becoming unresponsive after clicking the button to test the Automation.

![image](https://user-images.githubusercontent.com/74260856/155401176-d8f1079d-6d3a-4485-97b6-8e1ac3caa7e5.png)

**Reply by mjashanks** [reply]
Hmm - are you selecting back alot of rows at any point? 

Happy to look into it if you want to share an export of your app - either to here or support@budibase.com
